src/ansible_compat/prerun.py

```diff
@@ -0,0 +1,427 @@
+"""Utilities for configuring ansible runtime environment."""
+import json
+import logging
+import os
+import pathlib
+import re
+import subprocess
+import sys
+from functools import lru_cache
+from typing import Any, Dict, List, Optional, Tuple, Type, Union
+
+import packaging
+import tenacity
+from packaging import version
+
+from ansible_compat.config import (
+    ansible_collections_path,
+    collection_list,
+    options,
+    parse_ansible_version,
+)
+from ansible_compat.constants import (
+    ANSIBLE_DEFAULT_ROLES_PATH,
+    ANSIBLE_MIN_VERSION,
+    ANSIBLE_MISSING_RC,
+    INVALID_CONFIG_RC,
+    INVALID_PREREQUISITES_RC,
+)
+from ansible_compat.loaders import yaml_from_file
+
+_logger = logging.getLogger(__name__)
+
+
+def check_ansible_presence(exit_on_error: bool = False) -> Tuple[str, str]:
+    """Assures we stop execution if Ansible is missing or outdated.
+
+    Returne found version and an optional exception if something wrong
+    was detected.
+    """
+
+    @lru_cache()
+    def _get_ver_err() -> Tuple[str, str]:
+
+        err = ""
+        failed = False
+        ver = ""
+        result = subprocess.run(
+            args=["ansible", "--version"],
+            stdout=subprocess.PIPE,
+            universal_newlines=True,
+            check=False,
+        )
+        if result.returncode != 0:
+            return (
+                ver,
+                "FATAL: Unable to retrieve ansible cli version: %s" % result.stdout,
+            )
+
+        ver, error = parse_ansible_version(result.stdout)
+        if error is not None:
+            return "", error
```

Comment:
Looks like golang here. We don't do this in Python. If you want to communicate an error, communicate it. Nothing prevents you from attaching the attributes to the exception.

Reply:
Added to https://github.com/ansible-community/ansible-compat/issues/6